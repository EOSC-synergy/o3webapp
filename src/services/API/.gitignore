# .gitignore in generated-client gets overwritten every run, thus putting this here instead
generated-client/.npmignore
generated-client/git_push.sh
generated-client/.openapi-generator-ignore
generated-client/.openapi-generator
openapitools.json
# gitignore the spec?
swagger.yml
