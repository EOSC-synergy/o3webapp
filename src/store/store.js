import { applyMiddleware, combineReducers, configureStore, createStore } from '@reduxjs/toolkit';
import plotReducer from './plotSlice/plotSlice';
import modelsReducer, { setModelsOfModelGroup } from './modelsSlice/modelsSlice';
import referenceReducer from './referenceSlice/referenceSlice';
import apiReducer from '../services/API/apiSlice/apiSlice';
import { DEFAULT_MODEL_GROUP } from '../utils/constants';
import { HYDRATE, createWrapper } from 'next-redux-wrapper';
import thunkMiddleware from 'redux-thunk';

const bindMiddleware = (middleware) => {
    if (process.env.NODE_ENV !== 'production') {
        const { composeWithDevTools } = require('redux-devtools-extension');
        return composeWithDevTools(applyMiddleware(...middleware));
    }
    return applyMiddleware(...middleware);
};

const combinedReducer = combineReducers({
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
    api: apiReducer,
});

const reducer = (state, action) => {
    if (action.type === HYDRATE) {
        const nextState = {
            ...state, // use previous state
            ...action.payload, // apply delta from hydration
        };
        // if (state.count.count) nextState.count.count = state.count.count; // preserve count value on client side navigation
        return nextState;
    } else {
        return combinedReducer(state, action);
    }
};

let store;

const initStore = () => {
    store = createStore(reducer, bindMiddleware([thunkMiddleware]));
    return store;
};

export { store };

export const wrapper = createWrapper(initStore);

export function createTestStore() {
    const store = configureStore({
        reducer: combinedReducer,
    });
    store.dispatch(setModelsOfModelGroup(DEFAULT_MODEL_GROUP)); // fill with default group
    return store;
}
