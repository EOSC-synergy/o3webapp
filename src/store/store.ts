import { Action, combineReducers, configureStore, ThunkAction } from '@reduxjs/toolkit';
import plotReducer, { GlobalPlotState } from './plotSlice';
import modelsReducer, { GlobalModelState, setModelsOfModelGroup } from './modelsSlice';
import referenceReducer, { GlobalReferenceState } from './referenceSlice';
import apiReducer from '../services/API/apiSlice/apiSlice';
import { DEFAULT_MODEL_GROUP } from '../utils/constants';
import { createWrapper, HYDRATE } from 'next-redux-wrapper';
import thunkMiddleware from 'redux-thunk';
import { useDispatch, useStore } from 'react-redux';

type GlobalState =
    | undefined
    | (GlobalPlotState & GlobalModelState & GlobalReferenceState & { api: any });

const combinedReducer = combineReducers({
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
    api: apiReducer,
});

const reducer = (state: GlobalState, action: { type: string; payload: any }) => {
    if (action.type === HYDRATE) {
        // if (state.count.count) nextState.count.count = state.count.count; // preserve count value on client side navigation
        return {
            ...state, // use previous state
            ...action.payload, // apply delta from hydration
        };
    } else {
        return combinedReducer(state, action);
    }
};

//let store: ReturnType<typeof configureStore<GlobalState, >>;

const initStore = () => {
    //store = createStore(reducer, bindMiddleware([thunkMiddleware]));
    return configureStore({
        reducer,
        devTools: process.env.NODE_ENV !== 'production',
        middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(thunkMiddleware),
    });
};

export type AppStore = ReturnType<typeof initStore>;
export type AppState = ReturnType<AppStore['getState']>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>;

export type AppDispatch = AppStore['dispatch'];
export const useAppDispatch = () => useDispatch<AppDispatch>(); // Export a hook that can be reused to resolve types

export const useAppStore = () => useStore<AppState>();

export const wrapper = createWrapper(initStore);

export function createTestStore() {
    const store = initStore();
    store.dispatch(setModelsOfModelGroup(DEFAULT_MODEL_GROUP)); // fill with default group
    return store;
}
