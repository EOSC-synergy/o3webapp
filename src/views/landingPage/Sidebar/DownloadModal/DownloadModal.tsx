import React from 'react';
import {
    FormControl,
    InputLabel,
    MenuItem,
    Modal,
    Select,
    Typography,
    Card,
    Button,
    IconButton,
    CardContent,
} from '@mui/material';
import type { SelectChangeEvent } from '@mui/material';
import { useTheme } from '@mui/material/styles';
import PropTypes from 'prop-types';
import { downloadGraphAsPDF } from '../../../../services/downloading/pdf/pdfCreator';
import {
    downloadGraphAsCSV,
    downloadGraphAsPNG,
    downloadGraphAsSVG,
} from '../../../../services/downloading/otherFormats';
import { useSelector } from 'react-redux';
import { selectPlotId, selectPlotTitle } from '../../../../store/plotSlice/plotSlice';
import { selectActivePlotData } from '../../../../services/API/apiSlice/apiSlice';
import { selectAllModelGroups } from '../../../../store/modelsSlice/modelsSlice';
import { REQUEST_STATE } from '../../../../services/API/apiSlice/apiSlice';
import CloseIcon from '@mui/icons-material/Close';
import CardHeader from '@mui/material/CardHeader';
import CardActions from '@mui/material/CardActions';

/**
 * The file formats which can be selected in the dropdown menu.
 * @constant {Array}
 * @memberof DownloadModal
 */
const fileFormats = [Symbol('CSV'), Symbol('PDF'), Symbol('PNG'), Symbol('SVG')];

type DownloadModalProps = {
    isOpen: boolean;
    onClose: () => void;
    reportError?: (error: string) => void;
};

/**
 * Opens a modal where the user can select the file format and download the plot.
 * @component
 * @param isOpen whether modal should be visible
 * @param onClose handles closing the modal
 * @param reportError enabling to report an error
 * @returns {JSX.Element} a jsx containing a modal with a dropdown to choose the file type and a download button
 */
const DownloadModal: React.FC<DownloadModalProps> = ({ isOpen, onClose, reportError }) => {
    /**
     * An array containing all model groups.
     * @constant {Array}
     */
    const modelGroups = useSelector((state) => selectAllModelGroups(state));

    /**
     * The plot id of the graph (tco3_zm, tco3_return etc.).
     * @constant {string}
     */
    const plotId = useSelector(selectPlotId);

    /**
     * The active data of the current plot which contains information about the models.
     * @constant {Object}
     *
     */
    const activeData = useSelector((state) => selectActivePlotData(state, plotId));

    /**
     * The plot title which is shown above the graph.
     * @constant {string}
     */
    const plotTitle = useSelector(selectPlotTitle);

    /**
     * The selected file format.
     * @constant {string}
     */
    const [selectedFileFormat, setSelectedFileFormat] = React.useState('');

    /**
     * An object containing the information about the style of the DownloadModal.
     * @constant {Object}
     */
    const style = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        width: '30%',
        bgColor: useTheme().palette.grey[200],
        boxShadow: 24,
        p: 5,
    };

    /**
     * Handles the download of the plot, if the download button is clicked.
     * @function
     */
    const handleDownloadPlot = () => {
        if (selectedFileFormat === 'PDF') {
            downloadGraphAsPDF(plotId, plotTitle, modelGroups, activeData.data);
        } else if (selectedFileFormat === 'PNG') {
            downloadGraphAsPNG(plotTitle);
        } else if (selectedFileFormat === 'SVG') {
            downloadGraphAsSVG(plotTitle);
        } else if (selectedFileFormat === 'CSV') {
            downloadGraphAsCSV(plotTitle, plotId, reportError);
        }
    };

    /**
     * Changes the selected file format.
     *
     * @param {Object} event    The event object given by the dropdown menu.
     * @function
     */
    const changeFileFormat = (event: SelectChangeEvent) => {
        setSelectedFileFormat(event.target.value);
    };

    return (
        <Modal open={isOpen} onClose={onClose}>
            <Card sx={style}>
                <CardHeader
                    title="Download Plot"
                    action={
                        <IconButton
                            onClick={onClose}
                            aria-label="close"
                            data-testid="DownloadModal-close"
                        >
                            <CloseIcon />
                        </IconButton>
                    }
                />
                <CardContent>
                    <FormControl style={{ width: '100%', minWidth: 150 }}>
                        <InputLabel id="formatSelectLabel">Format</InputLabel>
                        <Select
                            labelId="formatSelectLabel"
                            id="formatSelection"
                            label="format"
                            value={selectedFileFormat}
                            onChange={changeFileFormat}
                            inputProps={{ 'data-testid': 'DownloadModal-select-file-format' }}
                        >
                            {fileFormats.map((elem, idx) => (
                                <MenuItem key={idx} value={elem.description}>
                                    {elem.description}
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                </CardContent>
                <CardActions sx={{ justifyContent: 'flex-end', marginTop: '2%' }}>
                    <Button
                        disabled={
                            selectedFileFormat === '' || activeData.status !== REQUEST_STATE.success
                        }
                        variant="contained"
                        onClick={handleDownloadPlot}
                        data-testid="DownloadModal-download-plot"
                    >
                        <Typography>Download</Typography>
                    </Button>
                </CardActions>
            </Card>
        </Modal>
    );
};

DownloadModal.propTypes = {
    /**
     * Tracks whether modal should be visible
     */
    isOpen: PropTypes.bool.isRequired,
    /**
     * Handles closing the modal
     */
    onClose: PropTypes.func.isRequired,
    /**
     * A function for error handling
     */
    reportError: PropTypes.func,
};

export default DownloadModal;
