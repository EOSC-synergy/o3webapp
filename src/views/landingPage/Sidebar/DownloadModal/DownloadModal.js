import React from "react";
import {
    FormControl,
    InputLabel,
    MenuItem,
    Modal,
    Select,
    Typography,
    Card,
    Button, IconButton, CardContent,
} from "@mui/material";
import { useTheme } from "@mui/material/styles";
import PropTypes from "prop-types";
import { fileFormats } from "../../../../utils/constants";
import { downloadGraphAsPDF} from "../../../../services/downloading/pdf/pdfCreator";
import { downloadGraphAsCSV, downloadGraphAsPNG, downloadGraphAsSVG} from "../../../../services/downloading/otherFormats";
import { useSelector } from "react-redux";
import { selectPlotId, selectPlotTitle} from "../../../../store/plotSlice/plotSlice";
import { selectActivePlotData} from "../../../../services/API/apiSlice";
import { selectAllModelGroups } from "../../../../store/modelsSlice/modelsSlice";
import { REQUEST_STATE } from "../../../../services/API/apiSlice";
import CloseIcon from "@mui/icons-material/Close";
import CardHeader from "@mui/material/CardHeader";
import CardActions from "@mui/material/CardActions";

/**
 * Opens a modal where the user can select the file format and download the plot.
 * @component
 * @param {Object} props
 * @param {boolean} props.isOpen -> whether modal should be visible
 * @param {function} props.onClose -> handles closing the modal
 * @param {function} props.reportError -> enabling to report an error
 * @returns {JSX.Element} a jsx containing a modal with a dropdown to choose the file type and a download button
 */
function DownloadModal(props) {

  /**
   * An array containing all model groups.
   */
  const modelGroups = useSelector(state => selectAllModelGroups(state));

  /**
   * The plot id of the graph (tco3_zm, tco3_return etc.).
   */
  const plotId = useSelector(selectPlotId);

  /**
   * The active data of the current plot which contains information about the models.
   */
  const activeData = useSelector(state => selectActivePlotData(state, plotId));

  /**
   * The plot title which is shown above the graph.
   */
  const plotTitle = useSelector(selectPlotTitle);

  /**
   * The selected file format.
   */
  const [selectedFileFormat, setSelectedFileFormat] = React.useState('');

  /**
   * The style of the DownloadModal.
   */
  const style = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: "30%",
    bgColor: useTheme().palette.grey[200],
    boxShadow: 24,
    p: 5,
  };

  /**
   * Handles the download of the plot, if the download button is clicked.
   */
  const handleDownloadPlot = () => {
    if (selectedFileFormat === "PDF") {
      downloadGraphAsPDF(plotId, plotTitle, modelGroups, activeData.data, props.reportError);
    } else if (selectedFileFormat === "PNG") {
      downloadGraphAsPNG(plotTitle, props.reportError);
    } else if (selectedFileFormat === "SVG") {
      downloadGraphAsSVG(plotTitle, props.reportError);
    } else if (selectedFileFormat === "CSV") {
      downloadGraphAsCSV(plotTitle, plotId, props.reportError);
    }
  };

  /**
   * Changes the selected file format in the Redux store.
   *
   * @param {event} event the event that called this function
   */
  const changeFileFormat = (event) => {
    setSelectedFileFormat(event.target.value);
  };

  return (
      <Modal open={props.isOpen} onClose={props.onClose}>
          <Card sx={style}>
              <CardHeader
                  title="Download Plot"
                  action={
                      <IconButton onClick={props.onClose} aria-label="close" data-testid="DownloadModal-close">
                          <CloseIcon />
                      </IconButton>
                  }
              />
              <CardContent>
                  <FormControl style={{ width: "100%", minWidth: 150 }}>
                      <InputLabel id="formatSelectLabel">Format</InputLabel>
                      <Select
                          labelId="formatSelectLabel"
                          id="formatSelection"
                          label="format"
                          value={selectedFileFormat}
                          onChange={changeFileFormat}
                          inputProps={{"data-testid":"DownloadModal-select-file-format"}}
                          
                      >
                          {fileFormats.map((elem, idx) => (
                              <MenuItem key={idx} value={elem.description}>
                                  {elem.description}
                              </MenuItem>
                          ))}
                      </Select>
                  </FormControl>
              </CardContent>
              <CardActions sx={{justifyContent: "flex-end", marginTop: "2%"}}>
                  <Button
                      disabled={selectedFileFormat === '' || activeData.status !== REQUEST_STATE.success}
                      variant="contained"
                      onClick={handleDownloadPlot}
                      data-testid="DownloadModal-download-plot"
                  >
                      <Typography>Download</Typography>
                  </Button>
              </CardActions>
          </Card>
      </Modal>
  );
}

DownloadModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  reportError: PropTypes.func,
};

export default DownloadModal;
