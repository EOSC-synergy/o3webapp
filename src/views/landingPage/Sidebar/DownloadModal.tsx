import React from 'react';
import {
    Button,
    Card,
    CardContent,
    FormControl,
    IconButton,
    InputLabel,
    MenuItem,
    Modal,
    Select,
    type SelectChangeEvent,
    Typography,
} from '@mui/material';
import { useTheme } from '@mui/material/styles';
import { downloadGraphAsPDF } from 'services/downloading/pdf/pdfCreator';
import {
    downloadGraphAsCSV,
    downloadGraphAsPNG,
    downloadGraphAsSVG,
} from 'services/downloading/otherFormats';
import { useSelector } from 'react-redux';
import { selectPlotId, selectPlotTitle } from 'store/plotSlice';
import { REQUEST_STATE, selectActivePlotData } from 'services/API/apiSlice';
import { type GlobalModelState, selectAllModelGroups } from 'store/modelsSlice';
import CloseIcon from '@mui/icons-material/Close';
import CardHeader from '@mui/material/CardHeader';
import CardActions from '@mui/material/CardActions';
import { type AppState } from 'store';
import { type ErrorReporter } from 'utils/reportError';

/**
 * The file formats which can be selected in the dropdown menu.
 *
 * @memberof DownloadModal
 * @constant {Array}
 */
const fileFormats = [Symbol('CSV'), Symbol('PDF'), Symbol('PNG'), Symbol('SVG')];

type DownloadModalProps = {
    isOpen: boolean;
    onClose: () => void;
    reportError?: ErrorReporter;
};

/**
 * Opens a modal where the user can select the file format and download the plot.
 *
 * @param isOpen Whether modal should be visible
 * @param onClose Handles closing the modal
 * @param reportError Enabling to report an error
 * @returns {JSX.Element} A jsx containing a modal with a dropdown to choose the file type and a
 *   download button
 * @component
 */
const DownloadModal: React.FC<DownloadModalProps> = ({ isOpen, onClose, reportError }) => {
    /**
     * An array containing all model groups.
     *
     * @constant {Array}
     */
    const modelGroups = useSelector((state: GlobalModelState) => selectAllModelGroups(state));

    /**
     * The plot id of the graph (tco3_zm, tco3_return etc.).
     *
     * @constant {string}
     */
    const plotId = useSelector(selectPlotId);

    /**
     * The active data of the current plot which contains information about the models.
     *
     * @constant {Object}
     */
    const activeData = useSelector((state: AppState) => selectActivePlotData(state, plotId));

    /**
     * The plot title which is shown above the graph.
     *
     * @constant {string}
     */
    const plotTitle = useSelector(selectPlotTitle);

    /**
     * The selected file format.
     *
     * @constant {string}
     */
    const [selectedFileFormat, setSelectedFileFormat] = React.useState('');

    /**
     * An object containing the information about the style of the DownloadModal.
     *
     * @constant {Object}
     */
    const style = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        width: '30%',
        bgColor: useTheme().palette.grey[200],
        boxShadow: 24,
        p: 5,
    };

    /**
     * Handles the download of the plot, if the download button is clicked.
     *
     * @function
     */
    const handleDownloadPlot = () => {
        if (activeData.status != REQUEST_STATE.success) {
            // TODO: error message
            return;
        }
        if (selectedFileFormat === 'PDF') {
            downloadGraphAsPDF(plotId, plotTitle, modelGroups, activeData.data);
        } else if (selectedFileFormat === 'PNG') {
            downloadGraphAsPNG(plotTitle);
        } else if (selectedFileFormat === 'SVG') {
            downloadGraphAsSVG(plotTitle);
        } else if (selectedFileFormat === 'CSV') {
            downloadGraphAsCSV(plotTitle, plotId, reportError);
        }
    };

    /**
     * Changes the selected file format.
     *
     * @function
     * @param {Object} event The event object given by the dropdown menu.
     */
    const changeFileFormat = (event: SelectChangeEvent) => {
        setSelectedFileFormat(event.target.value);
    };

    return (
        <Modal open={isOpen} onClose={onClose}>
            <Card sx={style}>
                <CardHeader
                    title="Download Plot"
                    action={
                        <IconButton
                            onClick={onClose}
                            aria-label="close"
                            data-testid="DownloadModal-close"
                        >
                            <CloseIcon />
                        </IconButton>
                    }
                />
                <CardContent>
                    <FormControl style={{ width: '100%', minWidth: 150 }}>
                        <InputLabel id="formatSelectLabel">Format</InputLabel>
                        <Select
                            labelId="formatSelectLabel"
                            id="formatSelection"
                            label="format"
                            value={selectedFileFormat}
                            onChange={changeFileFormat}
                            inputProps={{ 'data-testid': 'DownloadModal-select-file-format' }}
                        >
                            {fileFormats.map((elem, idx) => (
                                <MenuItem key={idx} value={elem.description}>
                                    {elem.description}
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                </CardContent>
                <CardActions sx={{ justifyContent: 'flex-end', marginTop: '2%' }}>
                    <Button
                        disabled={
                            selectedFileFormat === '' || activeData.status !== REQUEST_STATE.success
                        }
                        variant="contained"
                        onClick={handleDownloadPlot}
                        data-testid="DownloadModal-download-plot"
                    >
                        <Typography>Download</Typography>
                    </Button>
                </CardActions>
            </Card>
        </Modal>
    );
};

export default DownloadModal;
