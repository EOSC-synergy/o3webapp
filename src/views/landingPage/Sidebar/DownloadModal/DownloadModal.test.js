import { render, fireEvent , within} from '@testing-library/react';
import '@testing-library/jest-dom';
import DownloadModal from './DownloadModal';
import { Provider } from "react-redux";
import { createTestStore } from '../../../../store/store';
import tco3zmResponse from "../../../../services/API/testing/tco3zm-response.json";
import axios from 'axios';
import { fetchPlotData } from '../../../../services/API/apiSlice';
import { O3AS_PLOTS } from '../../../../utils/constants';
jest.mock('axios');



describe('testing DownloadModal rendering', () => {
    let store;
    beforeEach(() => {
      store = createTestStore();
    });

    it('renders without crashing', () => {
        render(<Provider store ={store}> <DownloadModal reportError={() => {}} onClose={()=>{}} isOpen={true} /></Provider>);
    });

    it('renders correctly when open', () => {
        let { baseElement, container } = render(
            <Provider store ={store} > <DownloadModal isOpen={true} onClose={() => {}} reportError={() => {}} /> </Provider>
        );
        expect(baseElement).toMatchSnapshot(); 
        expect(container).toBeVisible();
    });

    it('renders correctly when closed', () => {
        let { container, baseElement } = render(
            <Provider store ={store}> <DownloadModal isOpen={false} onClose={() => {}} reportError={() => {}} /></Provider>
        );
        expect(baseElement).toMatchSnapshot();
        expect(container).not.toBeVisible;
    });

    it('raises a console.error function if a required prop is not provided', () => {
        console.error = jest.fn();
        render(
            <Provider store ={store}> <DownloadModal onClose={()=>{}} reportError={()=>{}} /></Provider>
        );
        expect(console.error).toHaveBeenCalled();
        render(
            <Provider store ={store}> <DownloadModal isOpen={true} reportError={() => {}} /></Provider>
        );
        expect(console.error).toHaveBeenCalled();
    });

    it('renders correctly when closed', async () => {
        
        const mock = jest.fn();
        let { getByTestId } = render(
            <Provider store ={store}> 
                <DownloadModal isOpen={true} onClose={() => {}} reportError={mock} />
            </Provider>
        );
        
        const wrap = getByTestId("DownloadModal-select-file-format")
        fireEvent.mouseDown(wrap);

        fireEvent.change(wrap, { target: { value: "CSV" } });
        fireEvent.click(getByTestId("DownloadModal-download-plot"));
        // mock api => data
        // simpler, but less powerful: axios.post.mockResolvedValue({data: tco3zmResponse});
        axios.post.mockImplementation((requestUrl) => {
            return Promise.resolve({data: tco3zmResponse})
        });
        await store.dispatch(fetchPlotData({plotId: O3AS_PLOTS.tco3_zm, models: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"]}))
        fireEvent.click(getByTestId("DownloadModal-download-plot"));
        expect(mock).toHaveBeenCalledWith("Can't download the chart if it hasn't been fully loaded.");
    });

});
