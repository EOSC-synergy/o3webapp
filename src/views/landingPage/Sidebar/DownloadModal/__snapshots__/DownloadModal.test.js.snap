// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing DownloadModal rendering renders correctly when closed 1`] = `
<body
  style=""
>
  <div>
     
  </div>
</body>
`;

exports[`testing DownloadModal rendering renders correctly when open 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
     
     
  </div>
  <div
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1on1d5p-MuiPaper-root-MuiCard-root"
      tabindex="-1"
    >
      <div
        class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
          >
            Download Plot
          </span>
        </div>
        <div
          class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
        >
          <button
            aria-label="close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="DownloadModal-close"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          style="width: 100%; min-width: 150px;"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="formatSelectLabel"
          >
            Format
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="formatSelectLabel formatSelection"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="formatSelection"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              data-testid="DownloadModal-select-file-format"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  format
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-19l6csj-MuiCardActions-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="DownloadModal-download-plot"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Download
          </p>
        </button>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
